cpp/src/arrow/acero/task_util.cc

```diff
@@ -328,10 +328,13 @@ Status TaskSchedulerImpl::ScheduleMore(size_t thread_id, int num_tasks_finished)
   }
 
   ARROW_DCHECK(register_finished_);
-
+#ifndef ARROW_ENABLE_THREADING
+    return ExecuteMore(thread_id, 1, true);
+#else
   if (use_sync_execution_) {
```

Comment:
Shouldn't instead `use_sync_execution_` to be always true when threading is disabled? @westonpace

Reply:
have done